indy_node/server/request_handlers/domain_req_handlers/attribute_handler.py

```diff
@@ -0,0 +1,72 @@
+from indy_common.state import domain
+
+from indy_common.constants import ATTRIB
+from plenum.common.constants import DOMAIN_LEDGER_ID, RAW, ENC, HASH, TARGET_NYM
+from plenum.common.exceptions import InvalidClientRequest, UnauthorizedClientRequest
+
+from plenum.common.request import Request
+from plenum.common.txn_util import get_type, get_request_data
+from plenum.server.database_manager import DatabaseManager
+from plenum.server.request_handlers.handler_interfaces.write_request_handler import WriteRequestHandler
+
+
+class AttributeHandler(WriteRequestHandler):
+
+    def __init__(self, database_manager: DatabaseManager):
+        super().__init__(database_manager, ATTRIB, DOMAIN_LEDGER_ID)
+
+    def static_validation(self, request: Request):
+        self._validate_type(request)
+        identifier, req_id, operation = get_request_data(request)
+
+        if not self._validate_attrib_keys(operation):
+            raise InvalidClientRequest(identifier, req_id,
+                                       '{} should have one and only one of '
+                                       '{}, {}, {}'
+                                       .format(ATTRIB, RAW, ENC, HASH))
+
+    def dynamic_validation(self, request: Request):
+        self._validate_type(request)
+        identifier, req_id, operation = get_request_data(request)
+
+        if not (not operation.get(TARGET_NYM) or
+                self.has_nym(operation[TARGET_NYM], isCommitted=False)):
+            raise InvalidClientRequest(identifier, req_id,
+                                       '{} should be added before adding '
+                                       'attribute for it'.
+                                       format(TARGET_NYM))
+
+        if operation.get(TARGET_NYM) and operation[TARGET_NYM] != identifier and \
+                not self.database_manager.idr_cache.getOwnerFor(operation[TARGET_NYM],
+                                                                isCommitted=False) == identifier:
+            raise UnauthorizedClientRequest(
+                identifier,
+                req_id,
+                "Only identity owner/guardian can add attribute "
+                "for that identity")
+
+    def gen_txn_path(self, txn):
+        path = domain.prepare_attr_for_state(txn, path_only=True)
+        return path.decode()
+
+    def _update_state_with_single_txn(self, txn, isCommitted=False) -> None:
```

Comment:
This needs to be `update_state`, or make a common `update_state` in `WriteReqHandler` that iterates through each txn and calls `_update_state_with_single_txn`.